relay-server/src/actors/spooler/mod.rs

```diff
@@ -1026,8 +1026,35 @@ mod tests {
         assert!(target_dir.exists());
     }
 
+    /// A helper that creates, uses and removes a temporary directory.
+    struct TempDirGuard {
+        path: PathBuf,
+        previous_path: PathBuf,
+    }
+
+    impl TempDirGuard {
+        fn new() -> Self {
+            let previous_path = std::env::current_dir().unwrap();
+            let path = std::env::temp_dir().join(Uuid::new_v4().to_string());
+            std::fs::create_dir(&path).unwrap();
+            std::env::set_current_dir(&path).unwrap();
+            Self {
+                path,
+                previous_path,
+            }
+        }
+    }
+
+    impl Drop for TempDirGuard {
+        fn drop(&mut self) {
+            std::env::set_current_dir(&self.previous_path).unwrap();
+            std::fs::remove_dir_all(&self.path).unwrap();
+        }
+    }
```

Comment:
Do we want to move this into a separate `testutils` module and then reuse it for other tests as well?

Reply:
Done. I also changed the helper to *not* automatically change the working directory, because it caused problems in parallel test runs: https://github.com/getsentry/relay/actions/runs/6666168668/job/18117173821?pr=2663#step:5:1661